mobile/lib/presentation/widgets/action_buttons/add_action_button.widget.dart

```diff
@@ -0,0 +1,201 @@
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/extensions/build_context_extensions.dart';
+import 'package:immich_mobile/presentation/widgets/action_buttons/base_action_button.widget.dart';
+import 'package:immich_mobile/providers/infrastructure/asset_viewer/current_asset.provider.dart';
+import 'package:immich_mobile/presentation/widgets/album/album_selector.widget.dart';
+import 'package:immich_mobile/providers/infrastructure/album.provider.dart';
+import 'package:immich_mobile/widgets/common/immich_toast.dart';
+import 'package:immich_mobile/providers/user.provider.dart';
+
+import 'package:immich_mobile/domain/models/album/album.model.dart';
+import 'package:immich_mobile/domain/models/asset/base_asset.model.dart';
+
+import '../../../constants/enums.dart';
+import 'archive_action_button.widget.dart';
+import 'move_to_lock_folder_action_button.widget.dart';
+
+enum _AddMenu { album, archive, lockedFolder }
+
+class AddActionButton extends ConsumerWidget {
+  const AddActionButton({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return Builder(
+      builder: (buttonContext) {
+        return BaseActionButton(
+          iconData: Icons.add,
+          label: "add_to".tr(),
+          onPressed: () => _showAddOptions(buttonContext, ref),
+        );
+      },
+    );
+  }
+
+  void _showAddOptions(BuildContext context, WidgetRef ref) {
+    final asset = ref.read(currentAssetNotifier);
+    if (asset == null) return;
+
+    final user = ref.read(currentUserProvider);
+    final isOwner = asset is RemoteAsset && asset.ownerId == user?.id;
+
+    final List<PopupMenuEntry<_AddMenu>> items = [
+      PopupMenuItem(
+        value: _AddMenu.album,
+        child: ListTile(leading: const Icon(Icons.photo_album_outlined), title: Text("album".tr())),
+      ),
+      if (asset.hasRemote && isOwner)
+        PopupMenuItem(
+          value: _AddMenu.archive,
+          child: ListTile(leading: const Icon(Icons.archive_outlined), title: Text("archive".tr())),
+        ),
+      if (asset.hasRemote && isOwner)
+        PopupMenuItem(
+          value: _AddMenu.lockedFolder,
+          child: ListTile(leading: const Icon(Icons.lock_outline), title: Text("locked_folder".tr())),
+        ),
+    ];
+
+    showMenu<_AddMenu>(
+      context: context,
+      color: context.themeData.scaffoldBackgroundColor,
+      position: _menuPosition(context),
+      items: items,
+    ).then((selected) {
+      if (selected == null) return;
+
+      // Safety re-check in case state changed while menu was open
+      final latest = ref.read(currentAssetNotifier);
+      final currentUser = ref.read(currentUserProvider);
+      final allowedOwner = latest is RemoteAsset && latest.ownerId == currentUser?.id;
+      final hasRemote = latest?.hasRemote == true;
```

Comment:
in which cases is this check needed? Swiping is locked when the context menu is opened and tapping off closes the menu so I dont think any of this is needed?